internal/pkg/manifest/validate.go

```diff
@@ -2236,7 +2245,6 @@ func (i ImageLocationOrBuild) validate() error {
 		return &errFieldMutualExclusive{
 			firstField:  "build",
 			secondField: "location",
-			mustExist:   true,
```

Comment:
wait aren't the mutually exclusive?

Reply:
I think we set `mustExist: true` if both `image.build` and `image.location` are empty. we are doing that validation over here. https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/manifest/validate.go#L714:L719 

In `errors.go` we have error message based on `mustExist` field. https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/manifest/errors.go#L94:L99

Reply:
interesting - should we just remove the check here then? is it just a duplicate validation?

Reply:
It is not a redundant validation. we are checking two cases.
1. if both `image.build` and `image.location` are empty
2. If both `image.build` and `image.location` are not empty.
I just moved both validation to one place in this commit [9fd2a4e](https://github.com/aws/copilot-cli/pull/5122/commits/9fd2a4e431c3ee2242411aeb97677f6b4012e3c4)